include/materials/norsand.tcc

```diff
@@ -94,6 +94,18 @@ mpm::dense_map mpm::NorSand<Tdim>::initialise_state_variables() {
   return state_vars;
 }
 
+//! Initialise state variables
+template <unsigned Tdim>
+std::vector<std::string> mpm::NorSand<Tdim>::state_variables() const {
+  const std::vector<std::string> state_vars = {
```

Comment:
I can't think of a better way actually. We might need to have both the vectors of state variables to get the key for the map of state variables where they are actually stored. Any chance the map is ordered so we don't need the vector (we are already using dense map)?

Reply:
Our dense map is open addressing, so we can have a contiguous array, we can't use ordered map as it will be stored as a tree rather than a contiguous array.

Reply:
Understood. OK, let's do it this way!